### ISSUE TITLE
MULTISET nested record as JSON_OBJECT() emulation (Db2, SQL Server) doesn't work for records with more than 10 fields

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13200]
When emulating `MULTISET` expressions using `JSON_OBJECT()`, we produce records of the form:

```
[{
  "v0": 1,
  "v1": true,
  "v2": null,
  "v3": "abc",
  ...
}]
```

We do this in Db2 (see #12105) and SQL Server (because SQL Server doesn't support `JSON_ARRAY`).

Now, the order of these object keys isn't guaranteed by JSON serialisation, even if we're using `LinkedHashMap` internally to represent them, so the `JSONReader` sorts the values by key. Unfortunately, this sort is merely lexical, meaning it will place `"v10"` between `"v1"` and `"v2"`.

**Comment by lukaseder** [comment]
This relies on the infrastructure added with https://github.com/jOOQ/jOOQ/issues/12930, a general set of fixes related to nesting rows and emulating them using the `MULTISET` emulation, rather than flattening them into the query.

As #12930 won't be backported, this fix here won't be backported either.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0
